File: spring-boot-sample-web/src/main/java/com/github/jccode/springbootsample/core/web/GlobalControllerExceptionHandler.java
package com.github.jccode.springbootsample.core.web;


import com.github.jccode.springbootsample.core.data.rest.RestResult;
import com.github.jccode.springbootsample.core.exception.RestException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;

import javax.servlet.http.HttpServletRequest;

@ControllerAdvice
public class GlobalControllerExceptionHandler {

    @ResponseBody
    @ExceptionHandler(RestException.class)
    public RestResult restException(Exception e, HttpServletRequest request) {
        return RestResult.fail(e.getMessage());
    }
}

File: spring-boot-sample-web/src/main/scala/com/github/jccode/springbootsample/core/web/GlobalControllerExceptionHandler.scala
package com.github.jccode.springbootsample.core.web

import javax.servlet.http.HttpServletRequest

import com.github.jccode.springbootsample.core.data.RestResult
import com.github.jccode.springbootsample.core.exception.RestException
import org.springframework.web.bind.annotation.{ControllerAdvice, ExceptionHandler, ResponseBody}

@ControllerAdvice
class GlobalControllerExceptionHandler {

  @ResponseBody
  @ExceptionHandler(Array(classOf[RestException]))
  def restException(e: Exception, request: HttpServletRequest): RestResult[Null, Null] =
    RestResult.fail(e.getMessage)
}
